packages/tools/examples/logicalOperators/index.ts

```diff
@@ -0,0 +1,437 @@
+  contourSegmentationToolBindings,
+} from '../../../../utils/demo/helpers';
+import { runAllPolylineTests } from '../../src/utilities/math/polyline/testPolylineOperations';
+
+// This is for debugging purposes
+console.warn(
+  'Click on index.ts to open source code for this example --------->'
+);
+
+const {
+  SplineContourSegmentationTool,
+
+  PlanarFreehandContourSegmentationTool,
+  ToolGroupManager,
+  Enums: csToolsEnums,
+  segmentation,
+} = cornerstoneTools;
+const { ViewportType } = Enums;
+
+// Define a unique id for the volume
+const toolGroupId = 'STACK_TOOLGROUP_ID';
+
+const segmentationId = `SEGMENTATION_ID`;
+const segmentIndexes = [1, 2, 3, 4, 5];
+const segmentVisibilityMap = new Map();
+let activeSegmentIndex = 0;
+const renderingEngineId = 'myRenderingEngine';
+
+// ======== Set up page ======== //
+
+setTitleAndDescription(
+  'Logical operations',
+  'Here we demonstrate how to use logical operations to combine two segments in a new segment'
+);
+
+const size = '500px';
+const content = document.getElementById('content');
+const viewportGrid = document.createElement('div');
+
+viewportGrid.style.display = 'flex';
+viewportGrid.style.flexDirection = 'row';
+
+const viewportId = 'CT_STACK_ACQUISITION';
+const element = document.createElement('div');
+
+element.oncontextmenu = () => false;
+element.style.width = size;
+element.style.height = size;
+
+viewportGrid.appendChild(element);
+
+content.appendChild(viewportGrid);
+
+function updateActiveSegmentIndex(segmentIndex: number): void {
+  activeSegmentIndex = segmentIndex;
+  segmentation.segmentIndex.setActiveSegmentIndex(segmentationId, segmentIndex);
+}
+
+function getSegmentsVisibilityState() {
+  let segmentsVisibility = segmentVisibilityMap.get(segmentationId);
+
+  if (!segmentsVisibility) {
+    segmentsVisibility = new Array(segmentIndexes.length + 1).fill(true);
+    segmentVisibilityMap.set(segmentationId, segmentsVisibility);
+  }
+
+  return segmentsVisibility;
+}
+
+function updateSegmentationConfig(config) {
+  segmentation.config.style.setStyle(
+    {
+      segmentationId,
+      type: csToolsEnums.SegmentationRepresentations.Contour,
+    },
+    config
+  );
+}
+
+// ============================= //
+
+const cancelDrawingEventListener = (evt) => {
+  const { element, key } = evt.detail;
+  if (key === 'Escape') {
+    cornerstoneTools.cancelActiveManipulations(element);
+  }
+};
+
+element.addEventListener(
+  csToolsEnums.Events.KEY_DOWN,
+  cancelDrawingEventListener
+);
+
+const Splines = {
+  CatmullRomSplineROI: {
+    splineType: SplineContourSegmentationTool.SplineTypes.CatmullRom,
+  },
+  LinearSplineROI: {
+    splineType: SplineContourSegmentationTool.SplineTypes.Linear,
+  },
+  BSplineROI: {
+    splineType: SplineContourSegmentationTool.SplineTypes.BSpline,
+  },
+};
+
+const SplineToolNames = Object.keys(Splines);
+const splineToolsNames = [...SplineToolNames];
+let selectedToolName = splineToolsNames[0];
+
+addDropdownToToolbar({
```

Comment:
there is actually a utility for this called utils/demo/helpers/addSegmentIndexDropdown.ts